### PR TITLE
Adds lots of catches, responses, and rebuttals

### PR DESCRIPTION
**Pull request by SpadesNeil**
Full list of changes:

- Added catch for "That's me" under positive responses to Birthday.
- Added response for "not giving you" to trigger the "Email Rebuttal.mp3"
- Added response for "don't really use" during the email question.
- Added rebuttal for "which do you have" during address question.
- Added rebuttal for "don't want insurance" to trigger "Nothing to be interested in.mp3" because when people say they "don't want insurance" they tend to have insurance and not want _new_ insurance.
- Added DNC catch for "take me off the call list".
- Added response for "who am I speaking" to trigger "CherylCalling.mp3"
- Added rebuttal for "insurance company are you" to trigger "What's LCN.mp3" since people misunderstand we're not an insurance _company_.
- Added catch for "don't have any vehicles" to trigger early wrapup.
- Added catches for "which do you have" and "which address" to trigger the "which address rebuttal.mp3"
- Added catches for "pardon" "huh" and "what" to trigger Cheryl to repeat herself.
- Changed "What do you have down" to "What do you have" for the information rebuttal.
- Fixed missing catches for Birthday.
- Fixed issues when asking how many vehicles someone has.
- Fixed variations of "how did you get" questions being missed.
- ~~Fixed "Do Not Call" being ignored in favor of "I'm not interested" by moving DNC up in the code. Previously, problems like, "I'm not interested, do not call" would trigger rebuttal when it shouldn't.~~ Nevermind. It works exactly as before.
- Fixed countless other missing words.
- Removed some small snips of redundant code here and there.

If/When this gets merged, **remember to squash-merge it, or you'll make the commit page look like a mess**.

### CODE REVIEW

**Comment by SpadesNeil** [comment]
1288 is testing this version.

**Comment by JTSwagger** [comment]
ran this from 12-230 no leads at all

File: AutoBotCSharp/Agent.cs [hunk 1]
```diff
@@ -3413,14 +3413,14 @@ public string badSpeechParser(string oldSpeech)
                     currentlyRebuttaling = true;
                     return false;
                 }
-                else if (resp.Contains("what did you say") || resp.Contains("what was that") || resp.Contains("what was the question")  || resp.Contains("do what") || resp.Contains("could you repeat that") || resp.Contains("come again") || resp == "excuse me" || resp.Contains("sorry I missed that"))
+                else if (resp.Contains("what did you say") || resp.Contains("what was that") || resp.Contains("what was the question")  || resp.Contains("do what") || resp.Contains("could you repeat that") || resp.Contains("come again") || resp == "excuse me" || resp.Contains("sorry I missed that") || resp.Contains("i'm sorry") || resp.Contains("say that again") || resp.Contains("I didn't catch that"))
```
Comment (SpadesNeil) [inline AutoBotCSharp/Agent.cs]: Need to correct the "I'm sorry" line. Contains() is the wrong format for this.

**Comment by SpadesNeil** [comment]
Corrected some glaring problems.

Uploading and running new test version to 1288 again, will run from 8:30 to the end of tonight's shift.

If someone audits these calls, please have them send me a copy of the error report so I can figure out what (if anything) is going wrong. Otherwise I'll audit them myself tomorrow.

**Comment by SpadesNeil** [comment]
Ok, I am comfortable with these changes. Removing Do Not Merge tag.

**Comment by SpadesNeil** [comment]
This bot is getting leads now. At the end of the week I'd like to see its performance report.

**Comment by SpadesNeil** [comment]
Updated and uploading to 1305 for testing.

**Comment by SpadesNeil** [comment]
Holding off on testing this until issues in the main branch are resolved.

**Comment by SpadesNeil** [comment]
This is updated with the current master. If the master doesn't tank, we can test this.

**Comment by SpadesNeil** [comment]
If the latest master build is getting leads, I'd like to test this on a few computers.

**Comment by SpadesNeil** [comment]
1299 has 10 leads entered as of 7PM
1290 also has 10
1276 has 9.

For comparison,
1280 has 11
1291 has 8
1297 has 6
1275 has 8

My bots are generally doing slightly better than most of the others today.

**Pull request merged**